module/Calendar.js

```diff
@@ -0,0 +1,311 @@
+					const timerId = setTimeout(
+						callback,
+						new Date(validatedDate).getTime() - Date.now()
+					);
+					console.log(callback);
+					events[index] = {
+						...events[index],
+						...validatedEvent,
+						timerId,
+						callback,
+					};
+				}
+				events[index] = {
+					...events[index],
+					...validatedEvent,
+					callback,
+				};
+				return 'Updated successfully!';
+			} else {
+				throw new Error('Event not found!');
+			}
+		} catch (error) {
+			console.log(error.message);
+		}
+	}
+
+	/**
+	 * The method returns array with events objects in the specified date range
+	 * @method getEventsForPeriod
+	 * @param {string} dateRange1
+	 * @param {string} dateRange2
+	 * @example
+	 * getEventsForPeriod('2022-10-10', '2022-10-20')
+	 */
+	getEventsForPeriod(dateRange1, dateRange2) {
+		try {
+			if (!dateRange1 && !dateRange2) {
+				throw new Error('Date range is required!');
+			}
+			if (isNaN(Date.parse(dateRange1)) && isNaN(Date.parse(dateRange2))) {
+				throw new Error('Incorrect date format!');
+			}
+			const parsedDateRange1 = Date.parse(dateRange1);
+			const parsedDateRange2 = Date.parse(dateRange2);
+			const events = this._events.filter((event) => {
+				return (
+					(Date.parse(event.date) >= parsedDateRange1 &&
+						Date.parse(event.date) <= parsedDateRange2) ||
+					(Date.parse(event.date) >= parsedDateRange2 &&
+						Date.parse(event.date) <= parsedDateRange1)
+				);
+			});
+			if (events.length) {
+				return events;
+			} else {
+				return 'There are no events for the specified period!';
+			}
+		} catch (error) {
+			console.log(error.message);
+		}
+	}
+
+	/**
+	 * The method return object with event in the specified date
+	 * @method getEventForDay
+	 * @param {string} date
+	 * @example
+	 * getEventForDay('2022-10-10')
+	 */
+	getEventForDay(date) {
+		try {
+			if (!date) {
+				throw new Error('Date is required!');
+			}
+			if (isNaN(Date.parse(date))) {
+				throw new Error('Incorrect date format!');
+			}
+			const parsedDate = Date.parse(date);
+			const events = this._events.filter((event) => {
+				return Date.parse(event.date) === parsedDate;
+			});
+			if (events.length) {
+				return events[0];
+			} else {
+				return 'There are no events for the specified period!';
+			}
+		} catch (error) {
+			console.log(error.message);
+		}
+	}
+
+	/**
+	 * The method return array with events objects in the specified month
+	 * @method getEventForMonth
+	 * @param {string|number} month number of month
+	 * @param {string|number} [year] number of year
+	 * @example
+	 * getEventForDay('04', '2022')
+	 * getEventForDay(4, 2022)
+	 * getEventForDay(4)
+	 */
+	getEventForMonth(month, year = new Date().getFullYear()) {
+		try {
+			const dateRange1 = year + '-' + month;
+			if (!month) {
+				throw new Error('Month is required!');
+			}
+			if (isNaN(Date.parse(dateRange1))) {
+				throw new Error('Incorrect date format!');
+			}
+			const nextMonth = (month) => {
```

Comment:
The code _smells_, a method declaration inside a method. There is also a lot of investment. Same throughout the code.